app/jobs/process_github_sponsor_update_job.rb

```diff
@@ -1,12 +1,19 @@
 class ProcessGithubSponsorUpdateJob < ApplicationJob
   queue_as :default
 
-  def perform(action, gh_username)
-    return unless action == 'created'
-
+  def perform(action, gh_username, node_id, privacy_level, is_one_time, monthly_price_in_cents)
     user = User.find_by(github_username: gh_username)
     return unless user
 
-    User::RegisterAsDonor.(user, Time.current)
+    case action
+    when 'cancelled'
+      Donations::Github::Sponsorship::HandleCancelled.(user, node_id, privacy_level, is_one_time, monthly_price_in_cents)
+    when 'created'
+      Donations::Github::Sponsorship::HandleCreated.(user, node_id, privacy_level, is_one_time, monthly_price_in_cents)
+    when 'edited'
+      Donations::Github::Sponsorship::HandleEdited.(user, node_id, privacy_level, is_one_time, monthly_price_in_cents)
+    when 'tier_changed'
+      Donations::Github::Sponsorship::HandleTierChanged.(user, node_id, privacy_level, is_one_time, monthly_price_in_cents)
+    end
```

Comment:
I'll cleanup unused parameters once we've decided what we want to do with the privacy level (see other comment)